### PR TITLE
add logs

### PR DESCRIPTION
**Pull request by pinzart**
I want to send this version of the Greg Client to Konrad so that we can have a better idea of the error codes returned by DPM

### CODE REVIEW

File: src/GregClient/GregClient.cs [hunk 1]
```diff
@@ -85,6 +87,40 @@ public ResponseWithContentBody<T> ExecuteAndDeserializeWithContent<T>(Request m)
...
+                if (!Directory.Exists(logDirPath))
+                {
+                    Directory.CreateDirectory(logDirPath);
+                }
+                using (StreamWriter outputFile = new StreamWriter(Path.Combine(logDirPath, Guid.NewGuid().ToString() + ".txt")))
+                {
+                    var logObj = new
+                    {
+                        timeStamp = DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss"),
+                        respContent = restResp.Content,
```
Comment (pinzart) [inline src/GregClient/GregClient.cs]: `respContent` should be public data right? nothing sensitive I hope..

Comment (gregmarr) [reply src/GregClient/GregClient.cs]: correct

**Comment by gregmarr** [comment]
The idea is sound. I'm not sure we want to commit this as-is. If we can send a local build that would be good.  Otherwise, we might want a flag to enable this (env var?).

File: src/GregClient/GregClient.cs [hunk 2]
```diff
@@ -85,6 +87,40 @@ public ResponseWithContentBody<T> ExecuteAndDeserializeWithContent<T>(Request m)
...
+                return;
+            }
+            try
+            {
+                var logDirPath = Path.Combine(Path.GetTempPath(), "GregLog");
+                if (!Directory.Exists(logDirPath))
+                {
+                    Directory.CreateDirectory(logDirPath);
+                }
+                using (StreamWriter outputFile = new StreamWriter(Path.Combine(logDirPath, Guid.NewGuid().ToString() + ".txt")))
```
Comment (gregmarr) [inline src/GregClient/GregClient.cs]: Should be a timestamp instead of a guid, for easier aligning with requests?

Comment (pinzart) [reply src/GregClient/GregClient.cs]: ok..might be more readable

Comment (pinzart) [reply src/GregClient/GregClient.cs]: My first thought was to try a config file for Greg.dll
`Greg.dll.config`:
```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
  <appSettings>
    <add key="debugLog" value="true" />
  </appSettings>
</configuration>
```
But I am not sure yet how to load such a config file...Trying to search for the MSDN  api now...
Any other ideas ? not sure how to make `env vars` work in this case...

Comment (pinzart) [reply src/GregClient/GregClient.cs]: Oh..I was thinking of process environment variables...
I am guessing you meant system wide env vars..that might work too

**Comment by pinzart** [comment]
@mjkkirschner @gregmarr I would like to add this to master..
Thoughts ?

File: src/GregClient/GregClient.cs [hunk 3]
```diff
@@ -85,6 +87,43 @@ public ResponseWithContentBody<T> ExecuteAndDeserializeWithContent<T>(Request m)
 
             return new ResponseWithContent<T>(response).DeserializeWithContent();
         }
+
+        private void LogResponse(IRestResponse restResp)
+        {
+            try
+            {
+                var enableDebugLogs = Environment.GetEnvironmentVariable("DEBUG_LOGS_DYNAMO_GREG");
```
Comment (mjkkirschner) [inline src/GregClient/GregClient.cs]: feels more idiomatic to use a config I think - 
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/App.config
https://github.com/DynamoDS/Dynamo/blob/2ef67e17e77bd12067bd33cda90c9de7ef38097b/src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs#L519

what do you think?

I can't think of anywhere else we use environment variables in the dynamo code base.

**Pull request merged**